### PR TITLE
Add Some Engineering Teams to CODEOWNERS

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do? What is the motivation?

Connecting the dots on who is requested on reviews when updates to these doc pages are made.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

**Review by brett0000FF — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -92,3 +92,15 @@ content/en/real_user_monitoring/error_tracking/browser.md       @Datadog/rum-bro
 
 # Observability Pipelines
 static/resources/yaml/observability_pipelines/                  @DataDog/observability-pipelines @DataDog/vector @DataDog/vector-admin @DataDog/documentation
```
Comment (alai97) [inline .github/CODEOWNERS:94]:
```suggestion
static/resources/yaml/observability_pipelines/                  @DataDog/observability-pipelines @Datadog/observability-pipelines-worker @DataDog/documentation
```

**Comment by brett0000FF** [comment]
@alai97 - Looks like @Datadog/observability-pipelines-worker doesn't exist?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -91,4 +91,16 @@ content/en/logs/error_tracking/browser_and_mobile.md            @Datadog/rum-bro
 content/en/real_user_monitoring/error_tracking/browser.md       @Datadog/rum-browser @DataDog/documentation
 
 # Observability Pipelines
-static/resources/yaml/observability_pipelines/                  @DataDog/observability-pipelines @DataDog/vector @DataDog/vector-admin @DataDog/documentation
+static/resources/yaml/observability_pipelines/                  @DataDog/observability-pipelines @Datadog/observability-pipelines-worker @DataDog/documentation
```
Comment (maycmlee) [inline .github/CODEOWNERS:94]:
```suggestion
static/resources/yaml/observability_pipelines/                  @DataDog/observability-pipelines-worker @DataDog/documentation
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -91,4 +91,16 @@ content/en/logs/error_tracking/browser_and_mobile.md            @Datadog/rum-bro
 content/en/real_user_monitoring/error_tracking/browser.md       @Datadog/rum-browser @DataDog/documentation
 
 # Observability Pipelines
-static/resources/yaml/observability_pipelines/                  @DataDog/observability-pipelines @DataDog/vector @DataDog/vector-admin @DataDog/documentation
+static/resources/yaml/observability_pipelines/                  @DataDog/observability-pipelines-worker @DataDog/documentation
```
Comment (alai97) [inline .github/CODEOWNERS:94]:
```suggestion
static/resources/yaml/observability_pipelines/                  @DataDog/observability-pipelines-worker-admin  @DataDog/documentation
```

**Pull request merged**